src/App.test.tsx

```diff
@@ -1,9 +1,49 @@
-import React from 'react';
 import { render, screen } from '@testing-library/react';
 import App from './App';
+import userEvent from '@testing-library/user-event';
+import { Provider } from 'react-redux';
+import store from './store';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+const mockUseNavigate = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  useNavigate: () => mockUseNavigate,
+  __esModule__: true
+}))
+
+const setup = () => {
+  render(
+    <Provider store={store}>
+      <App />
+    </Provider>
+)};
+
+describe('App', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('logging in navigates to the dashboard', async () => {
```

Comment:
It might be worth adding a test that checks that `authUser` is dispatched (maybe implicitly by checking the store state to see if it has the username in it after submitting)